scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -237,9 +238,18 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)])
    */
   def hashJoin[W: ClassTag](that: SCollection[(K, W)])
   : SCollection[(K, (V, W))] = self.transform { in =>
-    val side = that.asMultiMapSideInput
+    val side = that.combine { case (k, v) =>
+      MMap(k -> ArrayBuffer(v))
```

Comment:
@nevillelyh @ravwojdyla I updated to use combine with mutable.Map and ArrayBuffer, what do you guys think? looks good?